### DISCUSSION TITLE
How can I make append the current date to the PDF's file name?

### DISCUSSION THREAD

**Discussion by putzwasser** [quarto-dev/quarto-cli#10732] [Q&A]
### Description

Hi there,

I have a quarto book project. To keep the versions I created, I'd like to append the current date to my generated PDFs.

I know that I can set a file name with the `output-file` option. Is there a way to make this dynamic?

Thanks for your input!

**Comment by cderv** [comment]
You could use a post render script that rename the output maybe after each render. 
https://quarto.org/docs/projects/scripts.html#pre-and-post-render

**Reply by AlbertoFabbri93** [reply]
The problem with this approach is that you get an error every time because Quarto can not find the output file:

```
ERROR: No output created by quarto render 02_filtering.qmd

Stack trace:
    at handleRenderResult (file:///Applications/RStudio.app/Contents/Resources/app/quarto/bin/quarto.js:91527:15)
    at renderForPreview (file:///Applications/RStudio.app/Contents/Resources/app/quarto/bin/quarto.js:91543:25)
    at eventLoopTick (ext:core/01_core.js:175:7)
    at async render (file:///Applications/RStudio.app/Contents/Resources/app/quarto/bin/quarto.js:91416:29)
GET: /Analysis/02_filtering.html (404: Not Found)
```
I am using the script posted [here](https://stackoverflow.com/a/77326658/11317720).

Is there a fix for this problem?

**Reply by mcanouil** [reply]
@AlbertoFabbri93 Use the variables described in the documentation shared by Christophe.

**Reply by AlbertoFabbri93** [reply]
@mcanouil This is the script that I am using (copied from stackoverflow):

`datestamp_output_file.R`:
```r
output_file <- Sys.getenv("QUARTO_PROJECT_OUTPUT_FILES")

file.rename(
  output_file,
  paste0(
    tools::file_path_sans_ext(output_file),
    format(Sys.time(), '-%Y-%m-%d'),
    '.',
    tools::file_ext(output_file))
)
```

and the yaml configuration:

`_quarto.yml`:
```yaml
project:
  post-render: ./scripts/datestamp_output_file.R
```

As you can see it is using one of the variables described in the documentation (the one that makes the most sense I would say). So, I am not really sure what to change. Can you give me another hint?

**Reply by mcanouil** [reply]
Can you provide a small and fully reproducible example?

**Reply by cderv** [reply]
> at renderForPreview (file:///Applications/RStudio.app/Contents/Resources/app/quarto/bin/quarto.js:91543:25)

@AlbertoFabbri93 this is when calling `quarto preview` right? 

I did not think of that. post-render will happen before preview launch, and probably preview will need to be aware of output file name change.

I'll check our code with this assumption. But please give us as much information as possible to be sure we understand the usage that lead to this error. 

I think I found it base on error message 
https://github.com/quarto-dev/quarto-cli/blob/7483c62fa9af0437f72b3341f9a1c53a22c58b66/src/command/preview/preview.ts#L390-L404

So I'll look into it but don't hesitate to share more. 

Thanks !

**Reply by cderv** [reply]
So indeed there is a limitation in my suggestion. 

Modifying output file in a post render script in a project is no compatible with previewing with Quarto. 

Running preview of website need to be done differently... 

We would need a mechanism to pass back the information on outfile for after the post render step. 

I'll open an issue about this

**Reply by AlbertoFabbri93** [reply]
Sorry for the lack of details, here is the repro:

`_quarto.yml`:

```r
project:
  title: "test_datastamp"
  post-render: ./datestamp_output_file.R
```

`test_datastamp.qmd`:

````r
---
title: "test_datastamp"
---

## Quarto

To learn more about Quarto see <https://quarto.org>.

```{r}
1 + 1
```
````

`datestamp_output_file.R`:

```r
output_file <- Sys.getenv("QUARTO_PROJECT_OUTPUT_FILES")

file.rename(
  output_file,
  paste0(
    tools::file_path_sans_ext(output_file),
    format(Sys.time(), '-%Y-%m-%d'),
    '.',
    tools::file_ext(output_file)
  )
)
```

And yes, the error happens when running `quarto preview test_datastamp.qmd`.

@cderv Can you link the bug report here once you have opened it?

**Reply by cderv** [reply]
- https://github.com/quarto-dev/quarto-cli/issues/12987
